### PR TITLE
Enhance Project and User Models, Add Smart Contract Functionality, and Update Controllers and Routes

### PR DESCRIPTION
**Pull request by Aradhya2708**
### This pull request introduces several key updates and new features to the project:
- Updated Models
- New Controllers
- Added Routes
- Added Smart Contracts and Web3 Utilities
- Added Github Utilities and helper functions

all new functionalities are not yet tested for all environments

### CODE REVIEW

File: server/src/models/projectModel.js [hunk 1]
```diff
@@ -1,36 +1,50 @@
...
+        type: Number,
+        required: true
+    },
+    prURLs: [{
+        type: String
+    }],
+    status: {
+        type: String,
+        enum: ['resolved', 'unresolved', 'not_started'],
+        default: 'started'
```
Comment (adit4443ya) [inline server/src/models/projectModel.js]: Make default to `not started`

**Comment by adit4443ya** [comment]
Just do these 2 changes quickly and we can wrap this PR

**Comment by adit4443ya** [comment]
Also what is exactly `project.applicationDate` representing
Is this showing the date of project creation or the date from when the project will started?

File: server/src/models/projectModel.js (hunk 1 above)
Comment (Aradhya2708) [reply server/src/models/projectModel.js]: Resolved this issue.

**Comment by Aradhya2708** [comment]
> Also what is exactly `project.applicationDate` representing Is this showing the date of project creation or the date from when the project will started?

Application Date is the date till which freelancers may apply to contribute. This is well known at time of project creation. Util routes may be made to edit this.

**Pull request merged**